app/controllers/lead_providers/your_schools_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module LeadProviders
+  class YourSchoolsController < ::LeadProviders::BaseController
+    before_action :set_lead_provider
+
+    def index
+      @cohorts ||= Cohort.where(start_year: [2021, 2022])
+      @selected_cohort = if params[:selected_cohort_id]
+                           Cohort.find(params[:selected_cohort_id])
+                         else
+                           Cohort.find_by(start_year: 2021)
+                         end
+
+      @school_search_form = SchoolSearchForm.new(*form_params_index)
+      @school_search_form.cohort_year = @selected_cohort.start_year
+      @school_search_form.lead_provider_id = @lead_provider.id
+      @school_search_form.partnership = "in_a_partnership"
+
+      @schools = @school_search_form.find_schools(params[:page])
+    end
+
+    def create
+      redirect_to lead_providers_your_schools_path(*form_params_post)
```

Comment:
Do we actually need this action if all it does is redirect to another action?

Reply:
It transforms the parameters I believe

Reply:
@kicferk1 wrote this originally iirc

Reply:
Indeed, I think it changes request params into url params. So it's bookmarkable.

Reply:
I'm open to suggestions - the reason why is because we use form_with with posts params - we use similar pattern elsewhere - however if you have an idea for refactoring form with - I am open to suggestions

Reply:
Would be neatly done with JS but we avoid JS like a plague 😃

Reply:
Marking as resolved through scientific dialog